rules/Naming/PropertyRenamer/PropertyPromotionRenamer.php

```diff
@@ -38,10 +38,6 @@ public function __construct(
 
     public function renamePropertyPromotion(Class_|Interface_ $classLike): void
     {
-        if (! $this->phpVersionProvider->isAtLeastPhpVersion(PhpVersionFeature::PROPERTY_PROMOTION)) {
-            return;
-        }
-
```

Comment:
please revert, check if php support property promotion is needed before rename

Reply:
@samsonasik  It's only an example resolution.
We need to have this condition commented to make changes like (tested on PHP 7.4):

Before
```php
    private ClassAnalyzer $analyzer ;
    public function __construct(
        ClassAnalyzer $analyzer
    ) {
        $this->analyzer = $analyzer;
    }
```
After
```php
    private ClassAnalyzer $classAnalyzer;
    public function __construct(
        ClassAnalyzer $classAnalyzer
    ) {
        $this->classAnalyzer = $classAnalyzer;
    }
```

Without changes in PropertyPromotionRenamer.php it gives me
```php
    private ClassAnalyzer $classAnalyzer;
    public function __construct(
        ClassAnalyzer $analyzer
    ) {
        $this->classAnalyzer = $analyzer;
    }
```

Reply:
that's may need different service, as this class is focus on rename property promotion, as it always a property.

On non-property promotion, passed param is not always fill the property value.

Reply:
Different service will look almost the same as `PropertyPromotionRenamer.php`. As you can see in tests it doesn't break anything besides property names. I'm using this solution on PHP 7.4 and it works fine. If you have a better solution feel free to make changes to this PR.

Reply:
Duplication is fine, as new service is check non-property promotion and will need check against property fetch assignment so that the different.

After everything patched, same logic can be moved to another service for reusability